COMPILING.md

```diff
@@ -1,85 +1,319 @@
-Instructions for compiling on a Debian-based system. The package names here are
-valid for Ubuntu 12.10 and may or may not work on your system.
+# General Linux Guide
 
-Building instructions, below, always assume you are running them from the
-Cataclysm:DDA source directory.
+To build Cataclysm from source you will need at least a C++ compiler, some basic developer tools, and necessary build dependencies. The exact package names vary greatly from distro to distro, so this part of the guide is intended to give you higher-level understanding of the process.
+
+## Compiler
+
+You have three major choices here: GCC, Clang and MXE.
+
+  * GCC is almost always the default on Linux systems so it's likely you already have it
+  * Clang is usually faster than GCC, so it's worth installing if you plan to keep up with the latest experimentals
+  * MXE is a cross-compiler, so of any importance only if you plan to compile for Windows on your Linux machine (not covered here)
```

Comment:
I read this as saying MXE builds aren't covered by the document, so imagine my surprise when I found MXE compile instructions further below.